includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -1630,4 +1657,26 @@ public function get_login_confirmation_settings() {
 			'redirect_url' => $redirect_url,
 		];
 	}
+
+	/**
+	 * Calculates taxes, based on a product and a particular price.
+	 *
+	 * @param WC_Product $product The product, for retrieval of tax classes.
+	 * @param float      $price   The price, which to calculate taxes for.
+	 * @return array              An array of final taxes.
+	 */
+	private function get_taxes( $product, $price ) {
+		if ( ! wc_tax_enabled() || ! $this->prices_exclude_tax() ) {
+			// Only proceed when taxes are enabled, but not included.
+			return [];
+		}
+
+		// Follows the way `WC_Cart_Totals::get_item_tax_rates()` works.
+		$tax_class = $product->get_tax_class();
+		$rates     = WC_Tax::get_rates( $tax_class );
+		// No cart item, `woocommerce_cart_totals_get_item_tax_rates` can't be applied here.
+
+		// Normally there should be a single tax, but `calc_tax` returns an array, let's use it.
+		return WC_Tax::calc_tax( $price, $rates, false );
```

Comment:
`calc_tax` says it returns `Array of rates + prices after tax.` Do we know in what cases we have multiple entries here?

Reply:
Not really :\ That's why I mentioned this in the PR description:

> ❓ One thing I'm not sure about is he fact that calc_tax() returns a numeric one-dimensional array. In my tests, it always had a single entry, which I used in my calculations, but when would there be more entries, and how should we treat them?

Reply:
Since we are looping over the array returned by this to add to the product price, might be a good idea to look into this a little, or if we can't reproduce a scenario then maybe picking the first entry from this array to always use that.

Reply:
This was interesting to test :)

So, after digging as low as the SQL in [`WC_Tax::get_matched_tax_rates()`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-tax.php#L350), I saw that the logic is:

1. Load the tax class of the product
2. Load a location, for this I think it is based either on the store address, or the logged in customer (not too relevant here)
3. Load all taxes, which apply to the given tax class and location
4. Pick one of those taxes per priority (this is a manually entered value, see the screenshot below)
5. Add all of the loaded taxes in ascending priority.
    - If the tax is normal, just calculate it based on product prices.
    - If the tax is compound, calculate it based on product prices, combined with all previous taxes.

In the end, the returned array can contain more than one line, but all of those should be added together in the end.

<img width="3008" alt="image" src="https://user-images.githubusercontent.com/5311119/187423089-bb143d06-9807-401f-b777-377b059d7023.png">